financial-connections/src/main/java/com/stripe/android/financialconnections/features/linkaccountpicker/LinkAccountPickerViewModel.kt

```diff
@@ -168,55 +179,137 @@ internal class LinkAccountPickerViewModel @Inject constructor(
-        val state = stateFlow.value
-        val payload = requireNotNull(state.payload())
-        val (account, _) =
-            requireNotNull(payload.accounts.first { it.first.id == state.selectedAccountId })
-        val nextPane = account.nextPaneOnSelection
-        // Caches the selected account.
-        updateCachedAccounts { listOf(account) }
-        eventTracker.track(
-            FinancialConnectionsAnalyticsEvent.AccountsSubmitted(
-                accountIds = setOf(account.id),
-                isSkipAccountSelection = false,
-                pane = PANE
+    fun onSelectAccountsClick() {
+        suspend {
+            val state = stateFlow.value
+            val payload = requireNotNull(state.payload())
+
+            val accounts = payload.selectedPartnerAccounts(state.selectedAccountIds)
+            updateCachedAccounts { accounts }
+
+            // We assume that at this point, all selected accounts have the same next pane.
+            // Otherwise, the user would have been presented with an update-required bottom
+            // sheet before.
+            val nextPane = accounts.lastOrNull()?.nextPaneOnSelection
+            val accountIds = accounts.map { it.id }.toSet()
+
+            eventTracker.track(Click("click.link_accounts", PANE))
+
+            eventTracker.track(
+                AccountsSubmitted(
+                    accountIds = accountIds,
+                    isSkipAccountSelection = false,
+                    pane = PANE
+                )
             )
-        )
-        when (nextPane) {
-            Pane.SUCCESS -> {
-                val activeInstitution = selectNetworkedAccounts(
+
+            if (nextPane == SUCCESS) {
+                selectAccounts(
                     consumerSessionClientSecret = payload.consumerSessionClientSecret,
-                    selectedAccountIds = setOf(account.id),
+                    accountIds = accountIds,
                 )
-                // Updates manifest active institution after account networked.
-                updateLocalManifest { it.copy(activeInstitution = activeInstitution.data.firstOrNull()) }
-                // Updates cached accounts with the one selected.
-                eventTracker.track(Click("click.link_accounts", PANE))
+            } else {
+                handleUnsupportedNextPane(nextPane)
             }
+        }.execute { copy(selectNetworkedAccountAsync = it) }
+    }
 
-            Pane.BANK_AUTH_REPAIR -> {
-                coreAuthorizationPendingNetworkingRepair.set(
-                    requireNotNull(payload.partnerToCoreAuths).getValue(account.authorization)
+    private suspend fun selectAccounts(
+        consumerSessionClientSecret: String,
+        accountIds: Set<String>,
+    ) {
+        selectNetworkedAccounts(
+            consumerSessionClientSecret = consumerSessionClientSecret,
+            selectedAccountIds = accountIds,
+        )
+        updateLocalManifest { it.copy(activeInstitution = null) }
```

Comment:
I think because we’re not sure what to use if it returns multiple institutions. But also, we’re heading straight to the success pane after this, so it doesn’t really matter. I’ll remove the call.